aten/src/ATen/native/cuda/fused_adam_utils.cuh

```diff
@@ -112,56 +112,59 @@ struct FusedAdamMathFunctor {
       int chunk_size,
       FusedOptimizerTensorListMetadata<depth>& tl,
       const float* lr_ptr,
-      const double lr,
-      const double beta1,
-      const double beta2,
-      const double weight_decay,
-      const double eps,
-      const bool maximize,
-      const bool amsgrad,
+      const double& lr,
+      const double& beta1,
+      const double& beta2,
+      const double& weight_decay,
+      const double& eps,
+      const bool& maximize,
       const float* grad_scale_ptr,
-      const float* found_inf_ptr,
-      const ADAM_MODE adam_mode) {
-    int tensor_loc = tl.block_to_tensor[blockIdx.x];
-    int chunk_idx = tl.block_to_chunk[blockIdx.x];
-    int n = tl.numel_for_tensor[tensor_loc];
-    double lr_double = lr_ptr ? *lr_ptr : lr;
+      const float* found_inf_ptr) {
+    const auto tensor_loc = tl.block_to_tensor[blockIdx.x];
+    const auto chunk_idx = tl.block_to_chunk[blockIdx.x];
+    const double lr_double = lr_ptr ? *lr_ptr : lr;
 
     if (found_inf_ptr && *found_inf_ptr == 1) {
       return;
     }
-    auto* step_count =
-        reinterpret_cast<const float*>(tl.state_steps_addresses[tensor_loc]);
+    const auto [bias_correction1, bias_correction2_sqrt] =
+        [&]() -> std::pair<double, double> {
+      auto* step_count =
```

Comment:
I had asked this in the bigggg PR; you can just answer once: does moving step_count calculation here minimize registers by allowing the step_count to go out of scope sooner?

Reply:
that's about my reasoning (or rationalization)

Reply:
This feels very much like something that shouldn't matter _that_ much. If you have time, it'd be nice to know how much this change affects the benchmark.